presto-main/src/main/java/com/facebook/presto/cost/CostComparator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.google.common.annotations.VisibleForTesting;
+
+import javax.inject.Inject;
+
+import java.util.Comparator;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class CostComparator
+{
+    private final double cpuWeight;
+    private final double memoryWeight;
+    private final double networkWeight;
+
+    @Inject
+    public CostComparator(FeaturesConfig featuresConfig)
+    {
+        this(featuresConfig.getCpuCostWeight(), featuresConfig.getMemoryCostWeight(), featuresConfig.getNetworkCostWeight());
+    }
+
+    @VisibleForTesting
+    public CostComparator(double cpuWeight, double memoryWeight, double networkWeight)
+    {
+        checkArgument(cpuWeight >= 0, "cpuWeight can not be negative");
+        checkArgument(memoryWeight >= 0, "memoryWeight can not be negative");
+        checkArgument(networkWeight >= 0, "networkWeight can not be negative");
+        this.cpuWeight = cpuWeight;
+        this.memoryWeight = memoryWeight;
+        this.networkWeight = networkWeight;
+    }
+
+    public Comparator<PlanNodeCostEstimate> forSession(Session session)
+    {
+        return (left, right) -> this.compare(session, left, right);
+    }
+
+    public int compare(Session session, PlanNodeCostEstimate left, PlanNodeCostEstimate right)
```

Comment:
why do you need session?

Reply:
Today, it's not needed, but it's quite natural that comparator could be configurable (like overriding weights). Requiring session now enforces that all callers provide it.